File: sigedu-frontend-final/src/api/estudianteService.js
import axios from './axios';

export const estudianteService = {
    getAll: async () => {
        const response = await axios.get('/estudiantes');
        return response.data;
    },
    getById: async (id) => {
        const response = await axios.get(`/estudiantes/${id}`);
        return response.data;
    },
    getByDni: async (dni) => {
        const response = await axios.get(`/estudiantes/dni/${dni}`);
        return response.data;
    },
    getByCodigo: async (codigo) => {
        const response = await axios.get(`/estudiantes/codigo/${codigo}`);
        return response.data;
    },
    buscar: async (nombre, apellido) => {
        const response = await axios.get(`/estudiantes/buscar?nombre=${nombre || ''}&apellido=${apellido || ''}`);
        return response.data;
    },
    create: async (estudiante) => {
        const response = await axios.post('/estudiantes', estudiante);
        return response.data;
    },
    update: async (id, estudiante) => {
        const response = await axios.put(`/estudiantes/${id}`, estudiante);
        return response.data;
    },
    delete: async (id) => {
        const response = await axios.delete(`/estudiantes/${id}`);
        return response.data;
    },
    reactivar: async (id) => {
        const response = await axios.patch(`/estudiantes/${id}/reactivar`);
        return response.data;
    },
    getActivos: async () => {
        const response = await axios.get('/estudiantes/activos');
        return response.data;
    }
};

File: sigedu-backend-final/src/main/java/com/sigedu/backend/service/EstudianteService.java
package com.sigedu.backend.service;

import com.sigedu.backend.entity.Estudiante;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.Optional;

public interface EstudianteService {
    List<Estudiante> findAll();
    Page<Estudiante> findAll(Pageable pageable);
    List<Estudiante> findAllActivos();  // NUEVO
    Optional<Estudiante> findById(Integer id);
    Optional<Estudiante> findByDni(String dni);
    Optional<Estudiante> findByCodigo(String codigo);
    Estudiante save(Estudiante estudiante);
    Estudiante update(Integer id, Estudiante estudiante);
    void delete(Integer id);
    void reactivar(Integer id);  // NUEVO
    List<Estudiante> buscarPorNombreOApellido(String nombre, String apellido);
    Page<Estudiante> buscarPaginado(String termino, Boolean estado, Pageable pageable);
}